File: Java/lab13_Proxy/task_2_1/src/RealProject.java
public class RealProject implements Project {
    private final String url;

    public RealProject(String url) {
        this.url = url;
        System.out.println("Loading project " + url + "...");
    }

    @Override
    public void run() {
        System.out.println("\nRunning project " + url + "...");
    }
}

File: Cpp/lab13_Proxy/task_2_1/src/real_project.cpp
#include "real_project.hpp"

#include <iostream>


RealProject::RealProject(std::string url)
: url(url)
{
  std::cout << "Loading project " << url << "..." << std::endl;
}

RealProject::~RealProject() {}

void RealProject::run()
{
  std::cout << "Running project " << url << "..." << std::endl;
}
